server/bleep/src/indexes/file.rs

```diff
@@ -44,26 +44,27 @@ struct Workload<'a> {
     repo_ref: String,
     repo_name: &'a str,
     repo_metadata: &'a RepoMetadata,
-    cache: &'a FileCache,
+    cache: &'a RepoCacheSnapshot,
     dir_entry: RepoDirEntry,
 }
 
 #[async_trait]
 impl Indexable for File {
-    fn index_repository(
+    async fn index_repository(
         &self,
         reporef: &RepoRef,
         repo: &Repository,
         repo_metadata: &RepoMetadata,
         writer: &IndexWriter,
         pipes: &SyncPipes,
     ) -> Result<()> {
-        let file_cache = repo.open_file_cache(&self.config.index_dir);
+        let file_cache = FileCache::new(&self.sql);
+        let repo_cache = file_cache.for_repo(reporef).await?;
```

Comment:
This is confusingly named as it's still a file cache. Can you rename to `file_cache_by_repo`